### PR TITLE
feat: support downloading car files from @helia/verified-fetch 

### PR DESCRIPTION
**Pull request by achingbrain**
Adds support for the `application/vnd.ipld.car` accept header to allow downloading CAR files of DAGs.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/verified-fetch/test/car.spec.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    })
+    expect(resp.status).to.equal(200)
+    expect(resp.headers.get('content-type')).to.equal('application/vnd.ipld.car; version=1')
+    expect(resp.headers.get('content-disposition')).to.equal(`attachment; filename="${cid.toString()}.car"`)
+    const buf = new Uint8Array(await resp.arrayBuffer())
+
+    expect(buf).to.equalBytes(await writer.bytes())
+  })
+
+  it('should support specify a filename for a CAR file', async () => {
```
Comment (SgtPooki) [inline packages/verified-fetch/test/car.spec.ts:49]:
```suggestion
  it('should support specifying a filename for a CAR file', async () => {
```

File: packages/verified-fetch/src/verified-fetch.ts [hunk 2]
```diff
@@ -157,7 +160,27 @@ export class VerifiedFetch {
...
+      .catch(err => {
+        this.log.error('could not write car', err)
+        stream.cancel(err)
+          .catch(err => {
+            this.log.error('could not cancel stream after car export error', err)
+          })
+      })
+
+    const response = okResponse(stream)
+    response.headers.set('content-type', 'application/vnd.ipld.car; version=1')
```
Comment (SgtPooki) [inline packages/verified-fetch/src/verified-fetch.ts:181]: Do we need to check/set the version of the car file we get/create?

https://specs.ipfs.tech/http-gateways/trustless-gateway/#car-version

File: packages/verified-fetch/src/verified-fetch.ts [hunk 3]
```diff
@@ -157,7 +160,27 @@ export class VerifiedFetch {
...
+
+    // write the DAG behind `cid` into the writer
+    c.export(cid, writer, options)
+      .catch(err => {
+        this.log.error('could not write car', err)
+        stream.cancel(err)
+          .catch(err => {
+            this.log.error('could not cancel stream after car export error', err)
+          })
+      })
```
Comment (SgtPooki) [inline packages/verified-fetch/src/verified-fetch.ts:178]: I'm not sure I could explain deeply what's going on here.

Could you add a jsdoc link to explainers for me and potential future devs?

File: packages/verified-fetch/src/verified-fetch.ts [hunk 4]
```diff
@@ -157,7 +160,27 @@ export class VerifiedFetch {
...
-    return notSupportedResponse('vnd.ipld.car support is not implemented')
+    const c = car(this.helia)
+    const { writer, out } = CarWriter.create(cid)
+
+    const stream = toBrowserReadableStream<Uint8Array>(async function * () {
+      yield * out
+    }())
+
+    // write the DAG behind `cid` into the writer
+    c.export(cid, writer, options)
```
Comment (SgtPooki) [inline packages/verified-fetch/src/verified-fetch.ts:171]: We may not want to pass full options here?

**Review by SgtPooki — APPROVED** [review]
Lgtm but a few comments and questions

File: packages/verified-fetch/src/verified-fetch.ts (hunk 2 above)
Comment (achingbrain) [reply packages/verified-fetch/src/verified-fetch.ts:181]: From what I can tell `@ipld/car`'s writer doesn't add the v2 pragma or append the index so it's only ever going to be a v1 car.  It also doesn't export a method that lets us obtain the version programatically.

I've opened this to double check: https://github.com/ipld/js-car/issues/161

File: packages/verified-fetch/src/verified-fetch.ts (hunk 3 above)
Comment (2color) [reply packages/verified-fetch/src/verified-fetch.ts:178]: Same 🙊.

Comment (achingbrain) [reply packages/verified-fetch/src/verified-fetch.ts]: Take a look now - I've simplified the code here and pushed the complexity into the `@helia/car` module which is arguably where it should have been in the first place.

https://github.com/ipfs/helia/pull/444 will need to be merged and this rebased before CI will pass.

Hopefully it's clearer now.

File: packages/verified-fetch/src/verified-fetch.ts (hunk 4 above)
Comment (achingbrain) [reply packages/verified-fetch/src/verified-fetch.ts]: We can change it if we need to, my experience has generally been that sometimes it's useful to be able to pass options to sub components.

As it stands there's complete overlap between the option types so I don't think all we'd gain by limiting the option keys is verbosity.

File: packages/verified-fetch/src/verified-fetch.ts (hunk 3 above)
Comment (SgtPooki) [reply packages/verified-fetch/src/verified-fetch.ts]: it's definitely better. Thanks!

**Review by SgtPooki — APPROVED** [review]
we have a few pending PRs which may change this PR further, but looks really good.

**Pull request merged**